src/app.js

```diff
@@ -181,10 +223,65 @@ gpii.app.exit = function (that) {
 };
 
 /**
-  * Handles when a user token is keyed out through other means besides the task tray key out feature.
-  * @param that {Component} An instance of gpii.app
-  * @param keyedOutUserToken {String} The token that was keyed out.
-  */
+ * Listens to any changes in the lifecycle managers logonChange model and calls the
+ * appropriate functions in gpii-app for notifying of the current user state
+ */
+gpii.app.logonChangeListener = function (that, model) {
+    if (model.inProgress === true) {
```

Comment:
I like to write (model.inProgress ? gpii.app.displayWaitDialog : gpii.app.dismissWaitDialog)(that); but some might consider this over-condensed